server/src/main/java/org/elasticsearch/action/admin/indices/create/CreatePartitionsRequest.java

```diff
@@ -21,62 +21,89 @@
 
 package org.elasticsearch.action.admin.indices.create;
 
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
 import org.elasticsearch.Version;
 import org.elasticsearch.action.support.master.AcknowledgedRequest;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
+import io.crate.metadata.PartitionName;
+import io.crate.metadata.RelationName;
 
 public class CreatePartitionsRequest extends AcknowledgedRequest<CreatePartitionsRequest> {
 
-    private final Collection<String> indices;
+    private final Collection<PartitionName> partitions;
 
-    /**
-     * Constructs a new request to create indices with the specified names.
-     */
-    public CreatePartitionsRequest(Collection<String> indices) {
-        this.indices = indices;
+    public CreatePartitionsRequest(Collection<PartitionName> partitions) {
```

Comment:
Could also change the representation here to have `RelationName` and `List<List<String>>` for partition values. 
This would restrict the request to create only partitions of a single table (which is how we use it currently) - that might make required changes in the transport a bit easier.

I'm also not sure if we should stick with `String` as type for the values. If we move away from the encoding scheme we also no longer have this requirement and could go with typed values.
But also not sure how much of that we should do now. I don't want to blow up the scope further but I also don't want to do multiple structures changes.